lib/Producer.js

```diff
@@ -100,39 +105,50 @@ class Producer {
     }
 
     if (!this.conns || this.conns.length === 0) {
-      return Promise.reject(new Error('No connections to nsqd'))
-        .asCallback(callback);
+      return Promise.reject(new Error('No connections to nsqd')).asCallback(callback);
     }
 
     if (options.retry && this.strategy === Producer.FAN_OUT) {
-      return Promise.reject(new Error('Retry on produce level is not supported in fanout strategy'))
-        .asCallback(callback);
+      return Promise.reject(new Error('Retry on produce level is not supported in fanout strategy')).asCallback(
+        callback
+      );
+    }
+
+    if (options.delay && parseInt(options.delay, 10) > 0) {
```

Comment:
why not use the term `defer`

Reply:
we've already wrapped up `writer.publish` by `Producer.produce` with self-define `options`, I'd like to make good use of it instead of creating another method with one more param.

Reply:
I'm talking about naming, in the README https://github.com/Wiredcraft/nsq-strategies#new-producerconnectconfig-option I refer nsqjs for other options so it's better to keep consistency.

Reply:
I'm fine with whichever makes it easier to understand.